precice/Dockerfile.Ubuntu1804.home

```diff
@@ -14,63 +21,63 @@ RUN apt-get -qq update && apt-get -qq install \
     git \
     python3-numpy \
     python3-dev \
+    python3-pip \
     petsc-dev \
     wget \
     bzip2 \
     cmake && \
     rm -rf /var/lib/apt/lists/*
 
-# create user precice
+# Create user precice
 ARG uid=1000
 ARG gid=1000
 RUN groupadd -g ${gid} precice \
-  && useradd -u ${uid} -g ${gid} -m -s /bin/bash precice
+    && useradd -u ${uid} -g ${gid} -m -s /bin/bash precice
 
-# set locale
+# Set locale
 RUN sed -i -e 's/# en_US.UTF-8 UTF-8/en_US.UTF-8 UTF-8/' /etc/locale.gen && \
     locale-gen
 ENV LANG en_US.UTF-8
 ENV LANGUAGE en_US:en
 ENV LC_NUMERIC en_US.UTF-8
 ENV LC_ALL en_US.UTF-8
 
-
-# Install PETSc from source
+# Install PETSc v3.12.5 from source (apt version too low)
 WORKDIR /petsc
 RUN wget -nv 'https://www.mcs.anl.gov/petsc/mirror/release-snapshots/petsc-3.12.5.tar.gz' -O - | tar xz && \
-  cd petsc-3.12.5 && \
-  ./configure && \
-  make all test
+cd petsc-3.12.5 && \
+./configure && \
+make all test
 RUN chown -R precice:precice /petsc
 
-# Rebuild image if force_rebuild after that command
+USER precice
+
+# Upgrade pip to latest version
+RUN pip3 install --user --upgrade pip
```

Comment:
If we already provide the latest version of pip with this image, why do we install pip (again?) [here](https://github.com/precice/systemtests/pull/260/files?file-filters%5B%5D=.calculix-adapter&file-filters%5B%5D=.codeaster-adapter&file-filters%5B%5D=.dealii-adapter_3D&file-filters%5B%5D=.fenics-adapter&file-filters%5B%5D=.home&file-filters%5B%5D=.mpich&file-filters%5B%5D=.openfoam-adapter&file-filters%5B%5D=.package&file-filters%5B%5D=.su2-adapter&file-filters%5B%5D=.yml#diff-3a5025fdfc7d5ffbc4d1140e2ef5014804df9f0e376db4c02177e3933de08b79L43)?

Reply:
If I understand it correctly `python3-pip` is no preCICE dependency. Therefore, I would suggest to **not** install it in `precice/Dockerfile.Ubuntu1234.xyz`, but where we actually need pip the first time (bindings, fenics-adapter, code-aster). This allows us to make sure that preCICE does not (by accident) depend on pip.

Reply:
I agree. In fact, it seems that `fenics-adapter` is currently the only adapter in need of pip. I removed the pip installs from the remaining precice dockerfiles and added the pip upgrade step to `fenics-adapter`.
It should be noted that if an adapter/test does require pip in the future, one should add `pip3 install --user --upgrade pip` as mentioned above to make sure that the version is sufficient at all times (currently a problem in Ubuntu 18.04, but not in 20.04).